### PR TITLE
TA-BB integration LTI middleware

### PR DESCRIPTION
**Pull request by todor996**
Fixed LTI middleware for TA-BB integration
ENV config for FE url and Redirect url

### CODE REVIEW

**Comment by ewpreston** [comment]
Some of these changes make sense but others are not generic at all. If we wanted the config in code we would have put the config in code, but we don't want the config in code. You can't just change the parameters to use when redirecting to the React UI without changing the React UI.

File: package.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
     "clsx": "^1.1.1",
     "cookie-parser": "^1.4.5",
     "core-js": "^3.18.2",
+    "dotenv": "^14.3.2",
```
Comment (ewpreston) [inline package.json:22]: where is this used?

File: server/src/app/routes.js [hunk 2]
```diff
@@ -178,14 +178,14 @@ module.exports = function (app) {
     // Before we can do that we need to get an authorization code for the current user.
     // Save off the JWT to our database so we can get it back after we get the auth code.
     const lmsServer = jwtPayload.body['https://purl.imsglobal.org/spec/lti/claim/tool_platform'].url;
-    const oneTimeSessionToken = jwtPayload.body['https://blackboard.com/lti/claim/one_time_session_token'];
+    const idToken = jwtPayload.body['https://blackboard.com/lti/claim/id_token'];
```
Comment (ewpreston) [inline server/src/app/routes.js:181]: why did you change this? Is that the value Learn sends now? Perhaps we had an old value of the claim.

File: server/src/app/routes.js [hunk 3]
```diff
@@ -233,7 +233,9 @@ module.exports = function (app) {
     } else if (jwtPayload.target_link_uri.endsWith('lti')) {
       res.redirect(`/lti_adv_view?nonce=${state}`);
     } else if (jwtPayload.target_link_uri.endsWith('lti13')) {
-      res.redirect(`/lti_adv_view?nonce=${state}`);
+      const jwtPayload = await db.getAuthFromState(state).jwt;
+      const uuid = jwtPayload.body['https://purl.imsglobal.org/spec/lti/claim/custom']['user_uuid'];
+      res.redirect(`${config.redirect_url}/${uuid}`);
```
Comment (ewpreston) [inline server/src/app/routes.js:238]: why are you sending user_uuid here instead of the nonce? We use that to look up the JWT from our cache

File: server/src/config/config.js [hunk 4]
```diff
@@ -1,28 +1,38 @@
...
- *
- */
-
-let overrides = '';
-
-let configResult = {};
-if (fs.existsSync('server/config/config_override.json')) {
-  overrides = require('../../config/config_override.json');
-  configResult = _.defaultsDeep(overrides, configJson);
+export default {
```
Comment (ewpreston) [inline server/src/config/config.js:1]: I don't want secrets and private keys in code. They don't belong in the code

File: server/src/app/routes.js (hunk 2 above)
Comment (todor996) [reply server/src/app/routes.js:181]: @ewpreston Yeah, they changed this i have no idea why, we got this information straight from them. For the rest of the changes, they are internal to our project and how we are handling it, i made a mistake when creating a PR and probably targetted the original instead of the forked one
